rye/src/cli/test.rs

```diff
@@ -67,7 +67,7 @@ pub fn execute(cmd: Args) -> Result<(), Error> {
         .join("pytest")
         .with_extension(EXE_EXTENSION);
 
-    let projects = locate_projects(project, cmd.all, &cmd.package[..])?;
+    let projects = locate_projects(project, cmd.all | cmd.package.is_empty(), &cmd.package[..])?;
```

Comment:
Covers 3 branches in the workspace case:

- `rye test --all`: all pyprojects are returned
- `rye test`: before would only returns the root -> now returns the same as `--all`
- `rye test -p <workspace_project>`: returns the matched projects

Reply:
I don't think that `rye test` should run all. It was intended to only run the tests of the package you are in. That is consistent with `fmt` and `lint`.

Reply:
I went by the cargo test implementation `https://doc.rust-lang.org/cargo/commands/cargo-test.html#package-selection`.

Although I see your point running pytests at the root directory would collect *ANY* test file such as exposed in #853 and #893

I see three options:

- Assume the default test directory is `tests` and allow the user to override with a `rye.test-directory` property
- Run all the packages in the workspace like `cargo test` (as implemented)
- Leave it to the user to configure `tool.pytest` like in #893